src/controller/python/chip/yaml/format_converter.py

```diff
@@ -71,6 +71,13 @@ def convert_yaml_type(field_value, field_type, use_from_dict=False):
     '''
     origin = typing.get_origin(field_type)
 
+    if isinstance(field_value, str) and field_value in config_values:
+        config_value = config_values[field_value]
+        if isinstance(config_value, dict) and 'defaultValue' in config_value:
```

Comment:
Just to confirm - is this handling cases where you can send in the config value directly (ie, config value matches the type of field type) or set config value to a dict with 'defaultValue' and 'type' to match the variable declaration system? Assuming that's what's happening here, two questions (that might not need to be addressed in this PR)
1) is there a 'type' entry in the config_values dict in this case? And does it need to be checked?
2) In the current system defaultValue can be overwritten on the command line. I realize you probably haven't gotten there yet, but will this design allow something similar?

Reply:
> 1. is there a 'type' entry in the config_values dict in this case? And does it need to be checked?

There are `types` in these entries. But currently I don't have a type string to a python type conversion. I am taking this out, but in the follow up PR I will mention a TODO that we should eventually be able to do this

>2.In the current system defaultValue can be overwritten on the command line. I realize you probably haven't gotten there yet, but will this design allow something similar?

Yes this design will allow for that. Very early on this config dict is created. It will be a matter of taking those overrides.

Reply:
awesome. sounds good.